### PR TITLE
Reduce MQ buffer length and flowgraph wait queue length to 16

### PR DESCRIPTION
**Pull request by congqixia**
See also #26177
/kind improvement

### CODE REVIEW

File: pkg/util/paramtable/component_param.go [hunk 1]
```diff
@@ -2452,7 +2452,7 @@ func (p *dataNodeConfig) init(base *BaseTable) {
 	p.FlowGraphMaxQueueLength = ParamItem{
 		Key:          "dataNode.dataSync.flowGraph.maxQueueLength",
 		Version:      "2.0.0",
-		DefaultValue: "1024",
+		DefaultValue: "16",
```
Comment (XuanYang-cn) [inline pkg/util/paramtable/component_param.go:2455]: it's 1024 in milvus.yaml, the same as pipeline of querynode, better to unify all DefaultValue to 16??
image
image

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]

File: pkg/mq/msgstream/mq_factory.go [hunk 2]
```diff
@@ -49,10 +49,11 @@ type PmsFactory struct {
 	PulsarNameSpace  string
 }
 
-func NewPmsFactory(config *paramtable.PulsarConfig) *PmsFactory {
+func NewPmsFactory(serviceParam *paramtable.ServiceParam) *PmsFactory {
+	config := &serviceParam.PulsarCfg
 	return &PmsFactory{
-		PulsarBufSize:    1024,
-		ReceiveBufSize:   1024,
+		PulsarBufSize:    serviceParam.MQCfg.MQBufSize.GetAsInt64(),
```
Comment (jiaoew1991) [inline pkg/mq/msgstream/mq_factory.go:55]: rename PulsarBufSize  to MQBufSize?

File: pkg/mq/msgstream/mq_factory.go [hunk 3]
```diff
@@ -180,23 +181,24 @@ func (f *KmsFactory) NewMsgStreamDisposer(ctx context.Context) func([]string, st
 	}
 }
 
-func NewKmsFactory(config *paramtable.KafkaConfig) Factory {
+func NewKmsFactory(config *paramtable.ServiceParam) Factory {
 	f := &KmsFactory{
 		dispatcherFactory: ProtoUDFactory{},
-		ReceiveBufSize:    1024,
-		config:            config,
+		ReceiveBufSize:    config.MQCfg.ReceiveBufSize.GetAsInt64(),
```
Comment (jiaoew1991) [inline pkg/mq/msgstream/mq_factory.go:187]: Does Kafka need the MQBufSize parameter?

**Pull request merged**